pkg/docker/docker.go

```diff
@@ -106,6 +106,13 @@ func CreateDockerConfigJSON(registryURL, username, password, targetPath, configP
 		}
 	}
 
+	if registryURL == "" || password == "" || username == "" {
+		if err := fileWrite(targetPath, dockerConfigContent, utils); err != nil {
+			return "", err
+		}
+		return targetPath, nil
+	}
+
```

Comment:
If the registryURL, password or username is not provided, it doesn't make sense to proceed further and add an empty entry to the Docker config.